### PR TITLE
Added Brand Test Runner support

### PR DESCRIPTION
**Pull request by najmsheikh**
### Goals :soccer:
To setup a framework that will allow us to automate much of the manual QA process for new integrations.

### Implementation :construction:
Added a `TestManager` class that has only one public method `parseIntent(~)`. This method will be called whenever `ButtonMerchant.trackIncomingIntent(~)` is called. The `TestManager` will parse the Intent for testing instructions. If found, it will execute the requested action and send the response **only** to the Brand Test Runner app. If the action is unsupported or if the app is not available on the device then no response will be sent anywhere.

### CODE REVIEW

File: button-merchant/src/androidTest/java/com/usebutton/merchant/TestManagerTest.java [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+        Intent intent = createIntentForAction(TestManager.ACTION_QUIT);
+        doThrow(ActivityNotFoundException.class).when(context).startActivity(any(Intent.class));
+
+        testManager.parseIntent(intent);
+
+        verify(terminator, never()).terminate();
+    }
+
+    @Test
+    public void parseIntent_getToken_shouldSendTokenInResponse() {
```
Comment (wessmith) [inline button-merchant/src/androidTest/java/com/usebutton/merchant/TestManagerTest.java]: I also tested that we get null if there is no token

File: button-merchant/src/androidTest/java/com/usebutton/merchant/TestManagerTest.java [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        when(buttonRepository.getSourceToken()).thenReturn("test-token");
+
+        testManager.parseIntent(intent);
+
+        Uri data = verifyBaseResponseStructure(TestManager.ACTION_GET_TOKEN);
+        assertEquals("test-token", data.getQueryParameter("btn_ref"));
+    }
+
+    @Test
+    public void parseIntent_postInstall_shouldSendResultInResponse() {
```
Comment (wessmith) [inline button-merchant/src/androidTest/java/com/usebutton/merchant/TestManagerTest.java:126]: I also tested that we get a `false` case if post install didn't get called

File: button-merchant/src/androidTest/java/com/usebutton/merchant/TestManagerTest.java [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+        assertEquals(Intent.FLAG_ACTIVITY_SINGLE_TOP, intent.getFlags());
+
+        Uri data = intent.getData();
+        assertNotNull(data);
+        assertEquals(TestManager.TEST_HARNESS_SCHEME, data.getScheme());
+        assertEquals(TestManager.ACTION_RESPONSE, data.getAuthority());
+        assertEquals("/" + actionType, data.getPath());
+
+        return data;
+    }
```
Comment (wessmith) [inline button-merchant/src/androidTest/java/com/usebutton/merchant/TestManagerTest.java:189]: Perhaps a test for an unrecognized intent it ignored

**Review by wessmith — APPROVED** [review]
LGTM -- noted a few more tests that could be added

**Review by chrismaddern — APPROVED** [review]

File: button-merchant/src/androidTest/java/com/usebutton/merchant/TestManagerTest.java (hunk 2 above)
Comment (najmsheikh) [reply button-merchant/src/androidTest/java/com/usebutton/merchant/TestManagerTest.java:126]: It's a primitive value so it'd be false anyway by default.
